addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlGeneratorUnitTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.graphql;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zaproxy.zap.testutils.TestUtils;
+
+public class GraphQlGeneratorUnitTest extends TestUtils {
+    GraphQlGenerator generator;
+    GraphQlParam param;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        setUpZap();
+        param = new GraphQlParam(5, 5, true, null, null);
+    }
+
+    @Test
+    public void scalarFieldsOnly() throws Exception {
+        generator = new GraphQlGenerator(getHtml("scalarFieldsOnly.graphql"), null, param);
+        String query = generator.generateFull(GraphQlGenerator.RequestType.QUERY);
+        query = query.replaceAll("\\s+", " ").trim();
+        String expectedQuery = "{ name id age height human }";
+        assertEquals(query, expectedQuery);
+    }
+
+    @Test
+    public void zeroDepthObjects() throws Exception {
+        generator = new GraphQlGenerator(getHtml("zeroDepthObjects.graphql"), null, param);
+        String query = generator.generateFull(GraphQlGenerator.RequestType.QUERY);
+        query = query.replaceAll("\\s+", " ").trim();
```

Comment:
t seems like this is being done in every test. Should it be being done in the code somewhere? Or does it indicate an issue with the generator?

Reply:
Seems like this isn't required anymore. I had put it there originally because the generator was adding spaces and newlines for pretty printing each query. But since the pretty printing is actually pointless (because the escape characters show as-is as part of a JSON string in the request editor), I had removed all of them, but forgot to update the tests. Good catch; I'll update the tests.

Reply:
Okay no worries.